end_to_end/3-mitigate-bias-train-model2-registry-e2e.ipynb

```diff
@@ -0,0 +1,949 @@
+{
```

Comment:
<p>The <code>train</code>  data contains <code>Nan</code>  values, and this code cell fails:</p><p> </p><pre class="ql-syntax" spellcheck="false">---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
&lt;ipython-input-74-f555ac526b6a&gt; in &lt;module&gt;
      1 sm = SMOTE(random_state=42)
----&gt; 2 train_data_upsampled, gender_res = sm.fit_resample(train, gender)
      3 train_data_upsampled['customer_gender_female'].value_counts()

/opt/conda/lib/python3.7/site-packages/imblearn/base.py in fit_resample(self, X, y)
     73             The corresponding label of <code>X_resampled</code>.
     74         """
---&gt; 75         check_classification_targets(y)
     76         arrays_transformer = ArraysTransformer(X, y)
     77         X, y, binarize_y = self._check_X_y(X, y)

/opt/conda/lib/python3.7/site-packages/sklearn/utils/multiclass.py in check_classification_targets(y)
    178     y : array-like
    179     """
--&gt; 180     y_type = type_of_target(y)
    181     if y_type not in ['binary', 'multiclass', 'multiclass-multioutput',
    182                       'multilabel-indicator', 'multilabel-sequences']:

/opt/conda/lib/python3.7/site-packages/sklearn/utils/multiclass.py in type_of_target(y)
    301     if y.dtype.kind == 'f' and np.any(y != y.astype(int)):
    302         # [.1, .2, 3] or [[.1, .2, 3]] or [[1., .2]] and not [1., 2., 3.]
--&gt; 303         _assert_all_finite(y)
    304         return 'continuous' + suffix
    305 

/opt/conda/lib/python3.7/site-packages/sklearn/utils/validation.py in _assert_all_finite(X, allow_nan, msg_dtype)
    104                     msg_err.format
    105                     (type_err,
--&gt; 106                      msg_dtype if msg_dtype is not None else X.dtype)
    107             )
    108     # for object dtype data, we only check for NaNs (GH-13254)

ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1974/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
resolved